src/Avalonia.X11/X11Window.cs

```diff
@@ -323,6 +329,49 @@ private void UpdateMotifHints()
                 PropertyMode.Replace, ref hints, 5);
         }
 
+        /// <summary>
+        /// Append `_NET_WM_PID` atom to X11 window
+        /// </summary>
+        /// <param name="windowXId"></param>
+        private void AppendPid(IntPtr windowXId)
+        {
+            // See https://github.com/AvaloniaUI/Avalonia/issues/17444
+            var pid = (uint)s_pid;
+            // The type of `_NET_WM_PID` is `CARDINAL` which is 32-bit unsigned integer, see https://specifications.freedesktop.org/wm-spec/1.3/ar01s05.html
+            XChangeProperty(_x11.Display, windowXId,
+                _x11.Atoms._NET_WM_PID, _x11.Atoms.XA_CARDINAL, 32,
+                PropertyMode.Replace, ref pid, 1);
+
+            // If _NET_WM_PID is set, the ICCCM-specified property WM_CLIENT_MACHINE MUST also be set.
+            // the hostname can change, so we can't cache it
+            // gethostname(3) on Linux just calls uname(2), so do it ourselves
```

Comment:
Why can't we just P/Invoke `gethostname` instead of that complex machinery in `GetUtsName`?

Reply:
I'd rather have an extra memcpy than complex piece of interop code.

Reply:
@kekekeks Address